### PR TITLE
Refactor containers

### PR DESCRIPTION
**Pull request by thibaultzanini**

### CODE REVIEW

File: ui/controls/docker-environment-input.reel/docker-environment-input.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        "owner": {
+            "properties": {
+                "element": {"#": "owner"},
+                "variableComponent": {"@": "variable"},
+                "valueComponent": {"@": "value"}
+            }
+        },
+
+        "memoryValidator": {
```
Comment (pchaussalet) [inline ui/controls/docker-environment-input.reel/docker-environment-input.html]: Badly named...

File: ui/controls/docker-environment-input.reel/docker-environment-input.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * @class DockerEnvironmentInput
+ * @extends AbstractMultipleSelectController
+ */
+exports.DockerEnvironmentInput = AbstractMultipleSelectController.specialize(/** @lends DockerEnvironmentInput# */ {
+
+    canAddNewEntryExpression: {
+        value: "!this.variableComponent.hasError && !this.valueComponent.hasError && !!this.variableComponent.value && !!this.valueComponent.value"
+    },
+
+    cleanCurrenEntry: {
```
Comment (pchaussalet) [inline ui/controls/docker-environment-input.reel/docker-environment-input.js]: Typo

File: ui/sections/containers/inspectors/docker-image-pull.reel/docker-image-pull.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * @module ui/docker-image-pull.reel
+ */
+var AbstractInspector = require("ui/abstract/abstract-inspector").AbstractInspector,
+    ContainerCreator = require("ui/sections/containers/inspectors/container-creator.reel").ContainerCreator;
+
+/**
+ * @class DockerImagePull
+ * @extends AbstractInspector
+ */
+var DockerImagePull = exports.DockerImagePull = AbstractInspector.specialize(/** @lends DockerImagePull# */ {
```
Comment (pchaussalet) [inline ui/sections/containers/inspectors/docker-image-pull.reel/docker-image-pull.js]: Why do you need to have the class as a variable ?

**Review by pchaussalet — CHANGES_REQUESTED** [review]

File: ui/controls/docker-port-input.reel/docker-port-input.html [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                "validator": {"@": "portValidator"}
+            }
+        },
+
+        "hostPort": {
+            "prototype": "blue-shark/ui/field-text-input.reel",
+            "properties": {
+                "element": {"#": "hostPort"},
+                "converter": {"@": "portConverter"},
+                "placeholderValue": ":8000",
```
Comment (pchaussalet) [inline ui/controls/docker-port-input.reel/docker-port-input.html]: Remove the `:`

File: ui/controls/docker-port-input.reel/docker-port-input.html [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                "label": "Host Port"
+            }
+        },
+
+        "containerPort": {
+            "prototype": "blue-shark/ui/field-text-input.reel",
+            "properties": {
+                "element": {"#": "containerPort"},
+                "converter": {"@": "portConverter"},
+                "placeholderValue": ":8000",
```
Comment (pchaussalet) [inline ui/controls/docker-port-input.reel/docker-port-input.html]: Remove the `:`

**Review by pchaussalet — CHANGES_REQUESTED** [review]

File: ui/sections/containers/inspectors/docker-image-pull.reel/docker-image-pull.js (hunk 3 above)
Comment (thibaultzanini) [reply ui/sections/containers/inspectors/docker-image-pull.reel/docker-image-pull.js]: dead code

**Review by pchaussalet — APPROVED** [review]
LGTM

**Pull request merged**